internal/telemetry/client.go

```diff
@@ -131,18 +138,37 @@ type Client struct {
 	// metrics are sent
 	metrics    map[string]*metric
 	newMetrics bool
+
+	// logLock gaurds the logging field
+	logLock sync.RWMutex
+	// Logging allows us to toggle on agentless in the case where
+	// there are issues with sending telemetry to the agent
+	Logging bool
 }
 
 func (c *Client) log(msg string, args ...interface{}) {
-	if c.Logger == nil {
+	// we don't log if the client is temporarily using agentless
+	// to avoid spamming the user with instrumentation telemetry error messages
+	c.logLock.RLock()
+	defer c.logLock.RUnlock()
+	if !c.Logging {
 		return
 	}
-	c.Logger.Printf(msg, args...)
+	log.Warn(fmt.Sprintf(LogPrefix+msg, args...))
+}
+
+// Started returns whether the client has started or not
+func (c *Client) Started() bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.started
 }
 
 // Start registers that the app has begun running with the given integrations
 // and configuration.
-func (c *Client) Start(integrations []Integration, configuration []Configuration) {
+// TODO (evan.li): errors is passed in here, but not actually used
```

Comment:
Do you mean _will_ be passed in here? or is it hidden inside the configuration?

Reply:
Yep, I left this TODO in a previous commit where `Start` accepted an error argument and then removed the argument later but forgot to update the comment, will update now